packages/orchestration/test/exos/local-chain-account-kit.test.ts

```diff
@@ -0,0 +1,162 @@
+    await bootstrap();
+
+  t.log('chainInfo mocked via `prepareMockChainInfo` until #8879');
+  const agoricChainInfo = prepareMockChainInfo(rootZone.subZone('chainInfo'));
+  const timestampHelper = makeTimestampHelper(timer, timer.getTimerBrand());
+
+  t.log('exo setup - prepareLocalChainAccountKit');
+  const baggage = makeScalarBigMapStore<string, unknown>('baggage', {
+    durable: true,
+  });
+  const { makeRecorderKit } = prepareRecorderKitMakers(baggage, marshaller);
+  const makeLocalChainAccountKit = prepareLocalChainAccountKit(
+    baggage,
+    makeRecorderKit,
+    // @ts-expect-error mocked zcf. use `stake-bld.contract.test.ts` to test LCA with offer
+    Far('MockZCF', {}),
+    timestampHelper,
+    agoricChainInfo,
+  );
+
+  t.log('request account from vat-localchain');
+  const lca = await E(localchain).makeAccount();
+  const address = await E(lca).getAddress();
+
+  t.log('make a LocalChainAccountKit');
+  const { holder: account } = makeLocalChainAccountKit({
+    account: lca,
+    address,
+    storageNode: makeMockChainStorageRoot().makeChildNode('lcaKit'),
+  });
+
+  t.truthy(account, 'account is returned');
+  t.regex(await E(account).getAddress(), /agoric1/);
+
+  const oneHundredStakeAmt = stake.make(1_000_000_000n);
+  const oneHundredStakePmt = issuerKit.mint.mintPayment(oneHundredStakeAmt);
+  const oneStakeAmt = stake.make(1_000_000n);
+
+  t.log('deposit 100 bld to account');
+  const depositResp = await E(account).deposit(oneHundredStakePmt);
+  t.true(AmountMath.isEqual(depositResp, oneHundredStakeAmt), 'deposit');
+
+  const destination: ChainAddress = {
+    chainId: 'cosmoslocal',
+    address: 'cosmos1pleab',
+    addressEncoding: 'bech32',
+  };
+
+  // TODO #9211, support ERTP amounts
+  t.log('ERTP Amounts not yet supported for AmountArg');
+  await t.throwsAsync(() => E(account).transfer(oneStakeAmt, destination), {
+    message: 'ERTP Amounts not yet supported',
+  });
+
+  t.log('.transfer() 1 bld to cosmos using DenomAmount');
+  const transferResp = await E(account).transfer(
+    { denom: 'ubld', value: 1_000_000n },
+    destination,
+  );
+  t.is(transferResp, undefined, 'Successful transfer returns Promise<void>.');
+
+  await t.throwsAsync(
+    () => E(account).transfer({ denom: 'ubld', value: 504n }, destination),
+    {
+      message: 'simulated unexpected MsgTransfer packet timeout',
+    },
+  );
+
+  const unknownDestination: ChainAddress = {
+    chainId: 'unknown',
+    address: 'uknown1pleab',
+    addressEncoding: 'bech32',
+  };
+  await t.throwsAsync(
+    () => E(account).transfer({ denom: 'ubld', value: 1n }, unknownDestination),
+    {
+      message: /not found/,
+    },
+    'cannot create transfer msg with unknown chainId',
+  );
+
+  await t.notThrowsAsync(
+    () =>
+      E(account).transfer({ denom: 'ubld', value: 10n }, destination, {
+        memo: 'hello',
+      }),
+    'can create transfer msg with memo',
```

Comment:
I wonder about unit tests for just building the msg. you had a file of message builders in cosgov. that might be handy

Reply:
Worth exploring. With jest I would reach for something like [toHaveBeenCalledWith](https://jestjs.io/docs/expect#tohavebeencalledwitharg1-arg2-) for on `toBridge`.

Reply:
You could use sinon to spy, but be [careful with shared state](https://www.github.com/avajs/ava/issues/560#issuecomment-185235049)

But it doesn't play well with SES and durability. When we need such capability we've been making fakes and putting in that instrumentation.